### PR TITLE
sql: introduce pg_extension schema

### PR DESCRIPTION
**Pull request by otan**
Also added a barebones geography_columns, geometry_columns and
spatial_ref_sys table definition which I will delegate to another
reviewer - this bit is SQL Feature-y by itself :)

Release note (sql change): Introduced the pg_extension virtual schema,
which contains tables which are on the public schema when using postgres
style extensions.

### CODE REVIEW

File: pkg/sql/logictest/testdata/logic_test/create_as [hunk 1]
```diff
@@ -132,7 +132,7 @@ query III colnames
 SELECT * FROM bar
 ----
 a  b  c
-1  2  4
+1  2  5
```
Comment (rohany) [inline pkg/sql/logictest/testdata/logic_test/create_as:135]: Why did this change?

File: pkg/sql/logictest/testdata/logic_test/geospatial [hunk 2]
```diff
@@ -103,7 +103,8 @@ INSERT INTO parse_test (geom, geog) VALUES
   (ST_GeomFromGeoJSON('{"type":"Feature","geometry":{"type":"Point","coordinates":[1,2]},"properties":null}'::jsonb), ST_GeogFromGeoJSON('{"type":"Feature","geometry":{"type":"Point","coordinates":[1,2]},"properties":null}'::jsonb)),
   (ST_GeomFromWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex')), ST_GeogFromWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex'))),
   (ST_GeomFromWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex'), 3857), ST_GeogFromWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex'), 3857)),
-  (ST_GeomFromEWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex')), ST_GeogFromEWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex')))
+  (ST_GeomFromEWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex')), ST_GeogFromEWKB(decode('0101000000000000000000F03F000000000000F03F', 'hex'))),
```
Comment (rohany) [inline pkg/sql/logictest/testdata/logic_test/geospatial:106]: are these changes part of adding pg_extension?

**Review by rohany — APPROVED** [review]

File: pkg/sql/logictest/testdata/logic_test/create_as (hunk 1 above)
Comment (otan) [reply pkg/sql/logictest/testdata/logic_test/create_as:135]: foo is `CREATE TABLE foo (x, y, z) AS SELECT catalog_name, schema_name, sql_path FROM information_schema.schemata`, and we added a new schema.

File: pkg/sql/logictest/testdata/logic_test/geospatial (hunk 2 above)
Comment (otan) [reply pkg/sql/logictest/testdata/logic_test/geospatial:106]: nah but it was an annoying rebase so i just moved this up from below.

**Comment by otan** [comment]
bors r=rohany

**Pull request merged**